crates/turbopack-node/src/source_map/content_source.rs

```diff
@@ -74,30 +89,23 @@ impl ContentSource for NextSourceMapTraceContentSource {
             }
         });
 
-        let wrapped = WrappedContentSourceVc::new(
-            self_vc.await?.asset_source,
-            NextSourceMapTraceContentProcessorVc::new(
-                id,
-                line,
-                column,
-                frame.name.map(|c| c.to_string()),
-            )
-            .into(),
-        );
-        Ok(ContentSourceResultVc::exact(
-            ContentSourceContent::Rewrite(
-                RewriteBuilder::new(file.path().to_string())
-                    .content_source(wrapped.as_content_source())
-                    .build(),
-            )
-            .cell()
-            .into(),
-        ))
-    }
-
-    #[turbo_tasks::function]
-    fn get_children(&self) -> ContentSourcesVc {
-        ContentSourcesVc::cell(vec![self.asset_source])
+        let sources = self_vc.await?.asset_source.get_routes().get(file.path());
+        let processor = NextSourceMapTraceContentProcessorVc::new(
+            id,
+            line,
+            column,
+            frame.name.map(|c| c.to_string()),
+        )
+        .into();
+        let sources = sources
+            .await?
+            .iter()
+            .map(|s| WrappedGetContentSourceContentVc::new(*s, processor).into())
+            .collect();
```

Comment:
Isn't this considerably more expensive, since we need to iterate a long list of catchalls, dynamics, fallbacks, etc?

Reply:
Yes, that's a problem. Luckily the list is usually short. But we can further optimize that in future. Eagerly returning the full list of `GetContentSourceContentVc`s isn't needed. Some kind of lazy iterator would be better.

Reply:
Leave a TODO with linear task, please.